fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java

```diff
@@ -7509,12 +7561,18 @@ public void onEraseOlapTable(OlapTable olapTable, boolean isReplay) {
                     int schemaHash = olapTable.getSchemaHashByIndexId(indexId);
                     for (Tablet tablet : materializedIndex.getTablets()) {
                         long tabletId = tablet.getId();
-                        List<Replica> replicas = ((LocalTablet) tablet).getReplicas();
-                        for (Replica replica : replicas) {
-                            long backendId = replica.getBackendId();
-                            DropReplicaTask dropTask = new DropReplicaTask(backendId, tabletId, schemaHash, true);
+                        if (partition.isUseStarOS()) {
+                            DropReplicaTask dropTask = new DropReplicaTask(
+                                    ((StarOSTablet) tablet).getPrimaryBackendId(), tabletId, schemaHash, true);
                             batchTask.addTask(dropTask);
-                        } // end for replicas
+                        } else {
+                            List<Replica> replicas = ((LocalTablet) tablet).getReplicas();
+                            for (Replica replica : replicas) {
+                                long backendId = replica.getBackendId();
+                                DropReplicaTask dropTask = new DropReplicaTask(backendId, tabletId, schemaHash, true);
```

Comment:
How to distinguish the staros tablet drop and the local tablet drop On be?

Reply:
Distinguish by storage medium and shard id.
I will add shard id later.